docs/migration-guides/prometheus-migration.md

```diff
@@ -0,0 +1,328 @@
+    ||| % [rsync_conf_b64],
+  ],
+  volumeMounts_+: {
+    [volume_name]: { mountPath: "/data" },
+  },
+};
+
+local containers_obj(containers) = {
+  [container.name]: container for container in containers
+};
+
+local stack = {
+  
+  target_with_sidecar: target {
+    spec+: {
+      template+: {
+        spec+: {
+          // convert containers array to obj for easier jsonnet
+          // handling: in particular, if we just added the new container
+          // to existing containers[] array, we wouldn't be able to re-run
+          // this, as we'd get duplicated containers.
+          // By referring the added container via rsync_container_name, we
+          // can easily override it.
+
+          local containers = super.containers,
+          containers: kube.PodSpec {
+            default_container:: containers[0].name,
+            // create or override(existing) rsync_container
+            containers_+: containers_obj(containers) {
+              [rsync_container_name]: rsync_container,
+              ["prometheus"]+:{
+                args+: [
+                  "--web.enable-admin-api",
+                ],
+              },
+            },
+          }.containers,
+        },
+      },
+    },
+  },
+};
+
+kube.List() {items_+: stack}
+```
+
+Notice also how one extra argument has been added to the Prometheus container. In order to generate a snapshot of Prometheus its mandatory to have the admin api enabled. Take a look at the [Prometheus docs](https://prometheus.io/docs/prometheus/latest/querying/api/#tsdb-admin-apis) for more information about this.
+
+After deploying the generated manifest you will have a rsync container that you can access with:
+
+`kubectl exec -it -n NAMESPACE PROMETHEUS_POD_NAME -c sidecar-rsync sh`
+
+> Note: Make sure you change the placeholders NAMESPACE and PROMETHEUS_POD_NAME for the Kubernetes namespace and Prometheus pod name you have. In example: `kubectl exec -it -n monitoring prometheus-d776d68fb-vc7gt -c sidecar-rsync sh`
+
+Check also that a new service has been created for Prometheus:
+
+```bash
+~ $ kubectl get svc
+NAME               TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+prometheus         ClusterIP   10.30.244.231   <none>        9090/TCP   1y
+prometheus-rsync   ClusterIP   10.30.247.115   <none>        873/TCP    9d
+```
+
+Now, you should be able to access to that service from other pod. If you are interested in doing a quick test from other pod you can do:
+
+```bash
+/ # nc -zv prometheus-rsync.NAMESPACE.svc.cluster.local 873
+prometheus-rsync.NAMESPACE.svc.cluster.local (10.130.32.5:873) open
+```
+
+## Modifying the `kubeprod-manifest.jsonnet` to add the sidecar rsync container
+
+The `kubeprod-manifest.jsonnet` file will contain all the custom configuration you want to apply to the jsonnet manifests.  In this case, in order to add the rsync sidecar container the following configuration will have to be added:
```

Comment:
will contain -> contains

jsonnet -> Jsonnet

"in order to add the rsync sidecar container, include the following configuration"

Reply:
done.